packages/babel-parser/src/parser/statement.js

```diff
@@ -880,8 +892,14 @@ export default class StatementParser extends ExpressionParser {
       }
 
       parsedNonDirective = true;
+      if (afterDirectivesParse) {
+        afterDirectivesParse(hasStrictDirective);
```

Comment:
This function should only be called once, the first time we find a non-directive:

```js
if (!parsedNonDirective && afterDirectivesParse) {
   afterDirectivesParse(hasStrictDirective);
}
parsedNonDirective = true
```